app/src/main/java/org/mozilla/reference/browser/BrowserApplication.kt

```diff
@@ -30,6 +30,11 @@ open class BrowserApplication : Application() {
     override fun onCreate() {
         super.onCreate()
 
+        // Set up Glean as early as possible in order to ensure availability to the crash reporting
+        // service.  Glean provides its own assurances that it is running on the main process so it
+        // should be safe calling the following function from another process.
+        setupGlean(this)
+
         setupCrashReporting(this)
```

Comment:
I didn't notice this before, but do we need to setup Glean before the crash reporter now? A big downside of that would be that any crash happening before the crash reporter setup would not get reported. So right now any crash in Glean setup or the experiments library would go unnoticed - that's very risky.

Reply:
Well, it doesn't need Glean initialized in order to _record_ the crash at all, just to record it _in Glean_.  That's what's happening in the init of the `GleanCrashReporterService`, if it's on the main process and Glean is initialized, then it checks for the crash(s) to be persisted in a file and records them in Glean.

Reply:
At least that was the idea... it's not ideal, I admit, but unfortunately Glean wasn't designed with multi-process at the front of our minds and so we relied upon some things that don't inter-process well.  Hopefully we can get around a little of that with the Rust core version?

Reply:
> 
> 
> At least that was the idea... it's not ideal, I admit, but unfortunately Glean wasn't designed with multi-process at the front of our minds and so we relied upon some things that don't inter-process well. Hopefully we can get around a little of that with the Rust core version?

Let me just a bit in here :-) Leaving processes out of the Glean SDK was a deliberate design choice: from the Glean perspective, processes are an implementation detail of the product itself.

Reply:
> 
> 
> I didn't notice this before, but do we need to setup Glean before the crash reporter now? A big downside of that would be that any crash happening before the crash reporter setup would not get reported. So right now any crash in Glean setup or the experiments library would go unnoticed - that's very risky.

You are correct, I didn't realize the implications of this as well. @pocmo , would you rather us to hold-back this PR and think of a solution for lib-crash?

Reply:
Solution for lib-crash/service-glean is inbound [here](https://github.com/mozilla-mobile/android-components/pull/3821).  Once that lands safely, then we should be safe to land this.  Note, I already restored the `onCreate()` function to the original which sets up crash reporting first.

Reply:
TL;DR of the fix is now libraries that consume Glean should no longer need to worry about whether Glean is initialized or not before recording metrics, which frees us up to initialize in whatever order we need to. 

I think there is still a small possibility to miss recording crash data into Glean if we crash before Glean is initialized, but I think that remains to be seen how often we run into this situation, and at least this way we don't need to have Glean initialized before lib-crash so we could still see Socorro or Sentry crashes to help us see if this is the case.